static/app/components/workflowEngine/form/control/priorityControl.tsx

```diff
@@ -0,0 +1,218 @@
+import {useCallback, useEffect, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {GroupPriorityBadge} from 'sentry/components/badge/groupPriority';
+import {Chevron} from 'sentry/components/chevron';
+import {CompactSelect, type SelectOption} from 'sentry/components/compactSelect';
+import {Flex} from 'sentry/components/container/flex';
+import FieldWrapper from 'sentry/components/forms/fieldGroup/fieldWrapper';
+import NumberField from 'sentry/components/forms/fields/numberField';
+import InteractionStateLayer from 'sentry/components/interactionStateLayer';
+import {IconArrow} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {PriorityLevel} from 'sentry/types/group';
+
+export interface PriorityControlGridProps {
+  name: string;
+  onChange?: (value: PriorityThresholds) => void;
+  value?: PriorityThresholds;
+}
+
+export interface PriorityThresholds {
+  created: PriorityLevel;
+  high?: number;
+  medium?: number;
+}
+
+export default function PriorityControl({
+  name,
+  value,
+  onChange,
+}: PriorityControlGridProps) {
+  const [thresholds, setThresholds] = useState<PriorityThresholds>(
+    value ?? {
+      created: PriorityLevel.LOW,
+      [PriorityLevel.MEDIUM]: 0,
```

Comment:
these two state objects feel like we might want to separate them; just thinking like "my current selected priority" is one state and "the value for the states" is another.